src/Core/Content/Product/SalesChannel/Review/ProductReviewLoader.php

```diff
@@ -0,0 +1,169 @@
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Sorting\FieldSorting;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Routing\Exception\MissingRequestParameterException;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+#[Package('inventory')]
+class ProductReviewLoader
+{
+    private const LIMIT = 10;
+    private const DEFAULT_PAGE = 1;
+    private const FILTER_LANGUAGE = 'filter-language';
+    private const DEFAULT_SORTING = 'createdAt';
+    private const ALLOWED_SORTINGS = ['createdAt', 'points'];
+
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly AbstractProductReviewRoute $route,
+        private readonly EventDispatcherInterface $eventDispatcher
+    ) {
+    }
+
+    /**
+     * load reviews for one product. The request must contain the productId
+     * otherwise MissingRequestParameterException is thrown
+     *
+     * @throws MissingRequestParameterException
+     * @throws \Shopware\Core\Framework\DataAbstractionLayer\Exception\InconsistentCriteriaIdsException
+     */
+    public function load(Request $request, SalesChannelContext $context): ProductReviewLoaderResult
+    {
+        $productId = $request->get('parentId') ?? $request->get('productId');
+        if (!$productId) {
+            throw new MissingRequestParameterException('productId');
+        }
+
+        $criteria = $this->createCriteria($request, $context);
+        $reviews = $this->route
+            ->load($productId, $request, $context, $criteria)
+            ->getResult()
+        ;
+
+        $reviewResult = ProductReviewLoaderResult::createFrom($reviews);
+        $reviewResult->setProductId($request->get('productId'));
+        $reviewResult->setParentId($request->get('parentId'));
+
+        $aggregation = $reviews->getAggregations()->get('ratingMatrix');
+        if ($aggregation instanceof TermsResult) {
+            $matrix = $aggregation->getBuckets();
+        }
+
+        $reviewResult->setMatrix(new RatingMatrix($matrix ?? []));
+        $reviewResult->setCustomerReview($this->getCustomerReview($productId, $context));
+        $reviewResult->setTotalReviews(
+            $reviewResult->getMatrix()->getTotalReviewCount()
+        );
+
+        $this->eventDispatcher->dispatch(new ProductReviewsLoadedEvent($reviewResult, $context, $request));
+
+        return $reviewResult;
+    }
+
+    private function createCriteria(Request $request, SalesChannelContext $context): Criteria
+    {
+        $limit = (int) $request->get('limit', self::LIMIT);
+        $page = (int) $request->get('p', self::DEFAULT_PAGE);
+        $offset = $limit * ($page - 1);
+
+        $sort = (string) $request->get('sort', self::DEFAULT_SORTING);
+        if (!\in_array($sort, self::ALLOWED_SORTINGS, true)) {
+            $sort = self::DEFAULT_SORTING;
```

Comment:
The case when an incorrect sorting value is declared in the request is not covered by the test. It would be great if you could cover this case.

Reply:
I added a test which has an invalid sorting: https://github.com/shopware/platform/pull/2389/files#diff-6b11a3de3d75954fd76a26118499db1e25887b9e4183cd326793289c6617c19bR178
And another test, which sets some other parameters for the request: https://github.com/shopware/platform/pull/2389/files#diff-6b11a3de3d75954fd76a26118499db1e25887b9e4183cd326793289c6617c19bR267